File: FMI/ANUL II/SEM 2/MPP/ProiectC#/Zboruri/Persistence/AngajatiDBRepository.cs
using System.Data;
using log4net;
using Model;

namespace Persistence;

public class AngajatiDbRepository : IAngajatiRepository
{
    private static readonly ILog Logger = LogManager.GetLogger("AngajatiDbRepository");
    private readonly IDictionary<string, string> _props = new SortedList<string, string>();

    public AngajatiDbRepository(IDictionary<string, string> props)
    {
        Logger.Info("Initializing AngajatiDBRepository");
        this._props["ConnectionString"] = "URI=" + props["ConnectionString"] + ", Version=3";
    }

    public void Save(Angajat entity)
    {
    }

    public void Update(string id, Angajat update)
    {
    }

    public Angajat FindOne(string id)
    {
        return null;
    }

    public IEnumerable<Angajat> FindAll()
    {
        return null;
    }

    public void Delete(string id)
    {
    }

    public Angajat? FindByUsernameAndPass(string user, string password)
    {
        Logger.InfoFormat("finding angajat with username {0}", user);
        IDbConnection con = DbUtils.GetConnection(_props);
        using (var comm = con.CreateCommand())
        {
            comm.CommandText = "select Username, Parola, Nume from angajati where Username=@user and Parola=@pass";
            IDbDataParameter paramUser = comm.CreateParameter();
            IDbDataParameter paramPass = comm.CreateParameter();
            paramUser.ParameterName = "@user";
            paramPass.ParameterName = "@pass";
            paramUser.Value = user;
            paramPass.Value = password;
            comm.Parameters.Add(paramUser);
            comm.Parameters.Add(paramPass);
        
            using (var dataR = comm.ExecuteReader())
            {
                if (dataR.Read())
                {
                    string username = dataR.GetString(0);
                    string parola = dataR.GetString(1);
                    string nume = dataR.GetString(2);
                    Angajat angajat = new Angajat(username, nume, parola)
                    {
                        Id = username
                    };
                    Logger.InfoFormat("Exiting findOne with value {0}", angajat);
                    return angajat;
                }
            }
        }
        
        Logger.InfoFormat("Exiting findOne with value {0}", null);
        return null;
    }
}

File: FMI/ANUL II/SEM 2/MPP/ProiectJava/Zboruri/ZboruriPersistence/src/main/java/ro/mpp2024/repository/AngajatiDBRepository.java
package ro.mpp2024.repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.mpp2024.domain.Angajat;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class AngajatiDBRepository implements AngajatiRepository {
    private final JdbcUtils dbUtils;
    private static final Logger logger = LogManager.getLogger();

    public AngajatiDBRepository(Properties props) {
        logger.info("Initializing AngajatiDBRepository with properties: {} ", props);
        dbUtils = new JdbcUtils(props);
    }

    @Override
    public int size() {
        return 0;
    }

    @Override
    public Angajat findOne(String s) {
        return null;
    }

    @Override
    public Iterable<Angajat> findAll() {
        return null;
    }

    @Override
    public void save(Angajat entity) {
    }

    @Override
    public void delete(String s) {
    }

    @Override
    public void update(String id, Angajat entity) {
    }

    @Override
    public Angajat findByUsernameAndPass(String username, String password) {
        logger.traceEntry("finding angajat with username {}", username);
        Connection connection = dbUtils.getConnection();
        try (PreparedStatement preparedStatement = connection.prepareStatement("select * from angajati where username = ? and parola = ?")) {
            preparedStatement.setString(1, username);
            preparedStatement.setString(2, password);
            try (ResultSet resultSet = preparedStatement.executeQuery()) {
                if (resultSet.next()) {
                    String nume = resultSet.getString("nume");
                    Angajat angajat = new Angajat(username, nume, password);
                    angajat.setId(username);
                    logger.traceExit();
                    return angajat;
                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB: " + e);
        }
        logger.traceExit("No angajat found with username {}", username);
        return null;
    }
}
